core/trino-main/src/main/java/io/trino/server/TrinoPluginManager.java

```diff
@@ -0,0 +1,264 @@
+        }
+
+        for (ResourceGroupConfigurationManagerFactory configurationManagerFactory : plugin.getResourceGroupConfigurationManagerFactories()) {
+            log.info("Registering resource group configuration manager %s", configurationManagerFactory.getName());
+            resourceGroupManager.addConfigurationManagerFactory(configurationManagerFactory);
+        }
+
+        for (SystemAccessControlFactory accessControlFactory : plugin.getSystemAccessControlFactories()) {
+            log.info("Registering system access control %s", accessControlFactory.getName());
+            accessControlManager.addSystemAccessControlFactory(accessControlFactory);
+        }
+
+        passwordAuthenticatorManager.ifPresent(authenticationManager -> {
+            for (PasswordAuthenticatorFactory authenticatorFactory : plugin.getPasswordAuthenticatorFactories()) {
+                log.info("Registering password authenticator %s", authenticatorFactory.getName());
+                authenticationManager.addPasswordAuthenticatorFactory(authenticatorFactory);
+            }
+        });
+
+        for (CertificateAuthenticatorFactory authenticatorFactory : plugin.getCertificateAuthenticatorFactories()) {
+            log.info("Registering certificate authenticator %s", authenticatorFactory.getName());
+            certificateAuthenticatorManager.addCertificateAuthenticatorFactory(authenticatorFactory);
+        }
+
+        headerAuthenticatorManager.ifPresent(authenticationManager -> {
+            for (HeaderAuthenticatorFactory authenticatorFactory : plugin.getHeaderAuthenticatorFactories()) {
+                log.info("Registering header authenticator %s", authenticatorFactory.getName());
+                authenticationManager.addHeaderAuthenticatorFactory(authenticatorFactory);
+            }
+        });
+
+        for (EventListenerFactory eventListenerFactory : plugin.getEventListenerFactories()) {
+            log.info("Registering event listener %s", eventListenerFactory.getName());
+            eventListenerManager.addEventListenerFactory(eventListenerFactory);
+        }
+
+        for (GroupProviderFactory groupProviderFactory : plugin.getGroupProviderFactories()) {
+            log.info("Registering group provider %s", groupProviderFactory.getName());
+            groupProviderManager.addGroupProviderFactory(groupProviderFactory);
+        }
+
+        for (ExchangeManagerFactory exchangeManagerFactory : plugin.getExchangeManagerFactories()) {
+            log.info("Registering exchange manager %s", exchangeManagerFactory.getName());
+            exchangeManagerRegistry.addExchangeManagerFactory(exchangeManagerFactory);
+        }
+    }
+
+    public static PluginClassLoader createClassLoader(String pluginName, List<URL> urls)
+    {
+        ClassLoader parent = PluginManager.class.getClassLoader();
+        return new PluginClassLoader(pluginName, urls, parent, SPI_PACKAGES);
+    }
```

Comment:
Can we have this as a part of `PluginManager` so that - other implementation need to implement a custom `Classloader`

Reply:
If we are planning to have it here - they can we modify this to `TrinoPluginManager.class.getClassLoader`

Reply:
I think moving it to `PluginManager` is a good idea.